flink-runtime/src/test/java/org/apache/flink/runtime/state/TestTaskStateManager.java

```diff
@@ -67,8 +69,11 @@ public TestTaskStateManager(
 		this.jobId = jobId;
 		this.executionAttemptID = executionAttemptID;
 		this.checkpointResponder = checkpointResponder;
-		this.taskStateSnapshotsByCheckpointId = new HashMap<>();
+		this.jobManagerTaskStateSnapshotsByCheckpointId = new HashMap<>();
+		this.taskManagerTaskStateSnapshotsByCheckpointId = new HashMap<>();
 		this.reportedCheckpointId = -1L;
+		this.subtaskLocalStateBaseDirectory =
+			new File(System.getProperty("java.io.tmpdir"), "testLocalState_" + UUID.randomUUID());
```

Comment:
Shouldn't this come from something like a `TemporaryFolder` such that the directory get's cleaned up in all cases. Consequently, maybe we should pass in the `subtaskLocalStateBaseDirectory` from the test using this class.

Reply:
This is already changed in later commits.